### PR TITLE
Update code to work with PHP 8.3

### PR DESCRIPTION
**Pull request by tthomsen**
I've tested this locally using compose install and the plugin is now downloading Avada plugins.

### CODE REVIEW

**Comment by szepeviktor** [comment]
Thank you!

File: src/ThemeFusionApi.php [hunk 1]
```diff
@@ -43,14 +43,12 @@ public function getPlugins(): array
     {
         $plugins = [];
 
-        $responseBody = $this->remoteFilesystem->getContents(
-            self::API_DOMAIN,
```
Comment (szepeviktor) [inline src/ThemeFusionApi.php:47]: Seems like API_DOMAIN is unused, please remove it.

File: src/ThemeFusionApi.php [hunk 2]
```diff
@@ -43,14 +43,12 @@ public function getPlugins(): array
...
-            self::API_BASE_URL . '/?' . \http_build_query(['avada_action' => 'get_plugins', 'avada_version' => $this->themeVersion]),
-            false // Hide progress indicator
+        $responseBody = $this->remoteFilesystem->get(
```
Comment (szepeviktor) [inline src/ThemeFusionApi.php]: please rename it to `$response` or `$apiResponse`.

File: src/ThemeFusionConfig.php [hunk 3]
```diff
@@ -22,7 +22,9 @@ class ThemeFusionConfig
 
     public function __construct(Config $composerConfig)
     {
-        $this->config = $composerConfig->get(self::THEMEFUSION_CONFIG);
+	    $this->config = $composerConfig->get(self::THEMEFUSION_CONFIG);
+	print("package config");
+	print_r($this->config);
```
Comment (szepeviktor) [inline src/ThemeFusionConfig.php]: Turmoil here!

File: src/ThemeFusionPackage.php [hunk 4]
```diff
@@ -100,4 +98,4 @@ public function isAbandoned()
     {
         return false;
     }
-}
+}
```
Comment (szepeviktor) [inline src/ThemeFusionPackage.php]: Please restore all final newlines.

File: src/ThemeFusionPlugin.php [hunk 5]
```diff
@@ -41,6 +41,7 @@ public function activate(Composer $composer, IOInterface $io): void
         $composerConfig = $composer->getConfig();
         $this->config = new ThemeFusionConfig($composerConfig);
         if (! $this->config->isValid()) {
+		    print("Invalid config");
```
Comment (szepeviktor) [inline src/ThemeFusionPlugin.php]: Indent with spaces only.
`error_log()` here please.

**Review by szepeviktor — CHANGES_REQUESTED** [review]

**Comment by szepeviktor** [comment]
What does PHPStan say?

**Review by tthomsen — COMMENTED** [review]
Okay this should fix the issue. If there is anything else let me know.

File: src/ThemeFusionApi.php [hunk 6]
```diff
@@ -82,31 +79,30 @@ public function getDownloadUrl(string $name): string
...
                 'item_name' => $name,
                 'ver' => $this->themeVersion,
-            ]),
-            false
+            ])
         );
 
-        if ($this->remoteFilesystem->findStatusCode($this->remoteFilesystem->getLastHeaders()) !== 200) {
+	if ($apiResponse->getStatusCode() !== 200) {
+		error_log('blah');
```
Comment (szepeviktor) [inline src/ThemeFusionApi.php]:
```suggestion
		error_log('Theme Fusion API error:'. (string)$apiResponse->getStatusCode());
```

**Review by tthomsen — COMMENTED** [review]
I think we are good now. Let me know if there is anything else

**Comment by szepeviktor** [comment]
Thank you.

**Comment by szepeviktor** [comment]
@tthomsen I've fixed a few problems reported by PHPStan.
https://github.com/szepeviktor/composer-theme-fusion/commit/e00ae318394e46516fba41a710852a2e9d1c8242

May I release it as 1.0?

**Pull request merged**